### ISSUE TITLE
Fails to read creation time for exited sessions on musl libc

### ISSUE THREAD

**Issue by pedromfedricci** [zellij-org/zellij#3114]
# 2. Issues with the Zellij UI / behavior / crash

## Basic information

`binary`: https://github.com/zellij-org/zellij/releases/download/v0.39.2/zellij-x86_64-unknown-linux-musl.tar.gz
`zellij --version`: zellij 0.39.2
`libc`: musl libc v1.1.24
`cross`: ghcr.io/cross-rs/x86_64-unknown-linux-musl:0.2.5

## Issue description

The recent new Session Resurrection features (CLI and Session Manager) fail to read the creation time (ctime) for exited sessions for musl targets since both Rust's std and libc do not support it. Libc only support [statx for gnu](https://github.com/rust-lang/libc/blob/main/src/unix/linux_like/linux/gnu/mod.rs#L19C1-L44C6) and so std does not provide [statx for musl](https://github.com/rust-lang/rust/blob/master/library/std/src/sys/pal/unix/fs.rs#L114C1-L134C2). Because of this, the **ReadAllSessionInfosOnMachine** background job will _always_ send the value 0 (Duration's default) as the current duration for exited sessions, [code location](https://github.com/zellij-org/zellij/blob/main/zellij-server/src/background_jobs.rs#L430-L452).

Zellij ls
![zellij-ls-0 39 2-musl](https://github.com/zellij-org/zellij/assets/12203476/5d2cfc3e-1c78-41d4-ac1f-c6e4f4789090)

Zellij session-manager
![zellij-session-manager-0 39 2-musl](https://github.com/zellij-org/zellij/assets/12203476/ba31bbd8-0b20-4b78-9789-64119e7ad128)

Since the **ReadAllSessionInfosOnMachine** background job fetches session data in a loop, this issue also happens to fill the log file with `Failed to read created stamp of resurrection file: Error { kind: Unsupported, message: "creation time is not available on this platform currently"` quite fast, for every tick (default 1sec) x no. of active sessions x no. of exited sessions with a session-layout.kdl file.

## Minimal reproduction

1. Download Zellij for linux/musl: https://github.com/zellij-org/zellij/releases/download/v0.39.2/zellij-x86_64-unknown-linux-musl.tar.gz
2. Run one or more clients with the download Zellij: `<your_path>/zellij`
3. Wait for about 1 minute so that Zellij will commit layout snapshots for you new sessions
4. Confirm that layout.kdl files have been written with: `ls ~/.cache/zellij/0.39.2/session_info/*`
5. Close all clients from 2
6. Check their exit status and **wrong** duration with `<your_path>/zellij ls` (created: 0s ago is incorrect)
7. Run a new client with download Zellij just like step 2
8. Check incoming new entries in the log file with: `tail -f /tmp/zellij-1000/zellij-log/zellij.log`

## Other relevant information

Building Zellij in a linux/gnu environment (`cargo build --release` or `cargo install zellij`) does not have such issues.

**Comment by pedromfedricci** [comment]
Some possible solutions that I could think of (other designs and ideas are welcome):

1. Keep current behavior for musl targets (always shows 0s ago) and either suppress the error entirely or cache it and then log it only once.
2. When targeting musl specifically, show the duration since last modification (that would be the last time Zellij took a snapshot) for exited sessions. This of course will not represent the "created at" duration accurately.
3. Update the behavior for all targets to show "last snapshot" instead of creation. The UI would read something like "Last Save", "Saved", etc. This would deviate the most from the current experience though. But one could argue that "last saved time" is a better hint to what the session will look like in comparison to its full duration.

**Comment by steakhutzeee** [comment]
I'm experiencing this too.

Also a question, so we need to wait 1 minute for the session to be cached?

If I create a session, issue some random command and then quit with Ctrl + q then I can't see the session with `zellij ls`

**Comment by pedromfedricci** [comment]
Yeah you need to wait for at least the **serialization_interval**'s value (default 60 seconds) and **session_serialization** must be set to **true** (which is the default value).

**Comment by steakhutzeee** [comment]
> Yeah you need to wait for at least the **serialization_interval**'s value (default 60 seconds) and **session_serialization** must be set to **true** (which is the default value).

Thanks! Can this time be configured? Do not see it in the docs.

**Comment by pedromfedricci** [comment]
It is not in the docs page yet, but you can add to your **config.kdl** like this (value is in seconds):
```kdl
serialization_interval 300
```

**Comment by brandonkal** [comment]
Could you release a gnu-linux binary at least until this is fixed?

**Comment by pedromfedricci** [comment]
I don't think a gnu based binary release is planned. The musl based one is convenient and easy enough to distribute, since it's all statically linked. The zellij package from your distribution is very likely built from source, targeting linux-gnu though. Another option is to build it yourself, I think that would only require you to install a Rust toolchain.

**Comment by sstark** [comment]
Is this also the reason why I get the following logged in zellij.log multiple times a second?

ERROR  |zellij_server::background| 2024-11-16 23:04:03.846 [async-std/runti] [zellij-server/src/background_jobs.rs:485]: Failed to read created stamp of resurrection file: Error { kind: Unsupported, message: "creation time is not available on this platform currently" }
ERROR  |zellij_server::background| 2024-11-16 23:04:03.846 [async-std/runti] [zellij-server/src/background_jobs.rs:485]: Failed to read created stamp of resurrection file: Error { kind: Unsupported, message: "creation time is not available on this platform currently" }
ERROR  |zellij_server::background| 2024-11-16 23:04:03.846 [async-std/runti] [zellij-server/src/background_jobs.rs:485]: Failed to read created stamp of resurrection file: Error { kind: Unsupported, message: "creation time is not available on this platform currently" }

**Comment by pedromfedricci** [comment]
> Is this also the reason why I get the following logged in zellij.log multiple times a second?
> 
> ```
> ERROR  |zellij_server::background| 2024-11-16 23:04:03.846 [async-std/runti] [zellij-server/src/background_jobs.rs:485]: Failed to read created stamp of resurrection file: Error { kind: Unsupported, message: "creation time is not available on this platform currently" }
> ERROR  |zellij_server::background| 2024-11-16 23:04:03.846 [async-std/runti] [zellij-server/src/background_jobs.rs:485]: Failed to read created stamp of resurrection file: Error { kind: Unsupported, message: "creation time is not available on this platform currently" }
> ERROR  |zellij_server::background| 2024-11-16 23:04:03.846 [async-std/runti] [zellij-server/src/background_jobs.rs:485]: Failed to read created stamp of resurrection file: Error { kind: Unsupported, message: "creation time is not available on this platform currently" }
> ```
Yeah, that is correct. I would recommend running a glibc linked executable for now, it will work properly.

**Comment by jul-sh** [comment]
Still experiencing this; any plans to release glibc linked binaries via gh releases?

**Comment by LilyKensa** [comment]
Can confirm in `Ubuntu 22.04.4 LTS x86_64`

All of my sessions that I created a few months ago show `[Created 0s ago]` when I run `zellij ls`

**Comment by iprok** [comment]
Zellij also spams the logs with the messages

> ERROR  |zellij_server::background| 2025-04-06 23:18:20.961 [async-std/runti] [zellij-server/src/background_jobs.rs:485]: Failed to read created stamp of resurrection file: Error { kind: Unsupported, message: "creation time is not available on this platform currently" }
> ERROR  |zellij_server::background| 2025-04-06 23:18:20.961 [async-std/runti] [zellij-server/src/background_jobs.rs:485]: Failed to read created stamp of resurrection file: Error { kind: Unsupported, message: "creation time is not available on this platform currently" }

Quickly making them several MB large.

**Comment by iprok** [comment]
Because of problem I have described previously I built zellij myself. You can also do it by using [my repository](https://github.com/iprok/zellij_builder) or download debs from it

**Comment by pedromfedricci** [comment]
Fixed by #4645, released under v0.44.0. Falls back to modification time in case creation time is not available.
